posts/_markdown/2023_qiskit_trebugger.md

```diff
@@ -0,0 +1,82 @@
+
+
+**Qiskit Trebugger (transpiler-debugger)** is a tool which provides a visual representation of the transpilation process. It was developed as an interactive tool with multiple views catering to the needs of different users. Available as a [python package on PyPI](https://pypi.org/project/qiskit-trebugger/), it can be used as a jupyter widget or  a lightweight CLI tool to understand how a circuit goes through the qiskit transpiler. The following image shows a preview of the CLI view - 
+
+<img src="../../images/2023_qiskit_trebugger_cli_full.png">
+
+The debugger is divided into three main components -
+
+**1. Overview of the Transpilation Process** 
+There are two panels which highlight crucial information about the transpilation process. Some basic information such as the quantum backend, optimization level and version information is specified as the first level of information. The second level contains the total number of passes executed, and a comparison of the different statistics for the original and final circuits.
+
+**2. Details of the Transpilation Process**
+The transpilation process in qiskit consists of multiple passes. Each of these passes either changes the circuit (Transformation) or the property set (Analysis). Moreover, the circuit properties such as its depth, width, operation count and type of gates change after each pass is executed. Both of the jupyter and CLI views provide a consolidated dashboard for this evolution.
+
+**3. Pass Level Details** 
+On a more granular level, users can also see a detailed view of each pass. This consists of the current circuit diagram, property set, logs emitted, documentation and the time of execution for the pass. 
+
+The `pip install qiskit-trebugger` command can be used to install the package in your python environment.
+
+### Views
+
+#### - Jupyter
+The debugger was originally built using the [ipywidgets](https://ipywidgets.readthedocs.io/en/stable/) package for the jupyter notebook environment. Whenever our package is used to debug a quantum circuit, an interactive widget is rendered consisting of the transpilation overview and the transpiler pass information. Some special features supported by this view are circuit diffs and property set hierarchy. Using our circuit diff feature, users can see what exactly changed between the circuits of two transpiler passes whereas an expandable property set is provided for users to look at its internals.
+
+**Running the `jupyter` view**
+
+```python
+
+from qiskit.providers.fake_provider import FakeCasablanca
+from qiskit.circuit.random import random_circuit 
+from qiskit_trebugger import Debugger
+import warnings
+
+warnings.simplefilter('ignore')
+debugger = Debugger(view_type = "jupyter")
+backend = FakeCasablanca()
+circuit = random_circuit(num_qubits = 4, depth = 5 , seed = 44)
+
+# replace qiskit's transpile call 
+debugger.debug(circuit, optimization_level = 2, backend = backend)
+```
```

Comment:
I am not sure how this will render in the html. In the past folks used gist (have a look at the unicorn/Quarc blog post, for example) to include code.